File: backend/app/schemas/audit.py
from datetime import datetime
from typing import Optional, List, Dict, Any
from pydantic import BaseModel, Field, ConfigDict

class AuditDefenseQueryRequest(BaseModel):
    question: str = Field(..., min_length=3)

class AuditDefenseQueryResponse(BaseModel):
    application_id: int
    question: str
    answer: str
    referenced_requirements: List[str] = []
    referenced_evidence_ids: List[int] = []
    referenced_overrides: List[Dict[str, Any]] = []
    assessment_finding: Optional[str] = None

class AuditOverrideRequest(BaseModel):
    requirement_id: Optional[int] = None
    original_status: str
    new_status: str
    reason: str = Field(..., min_length=5)

class AuditOverrideResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    application_id: int
    requirement_id: Optional[int] = None
    original_status: str
    new_status: Optional[str] = None
    reason: str
    hr_user_id: int
    created_at: datetime

File: frontend/src/api/audit.ts
import { apiClient } from './client';
import type { FinalAnalysisResponse } from './ai';

export interface AuditOverridePayload {
  requirement_id?: number | null;
  original_status?: string | null;
  new_status: string;
  reason: string;
}

export interface AuditOverrideResponse {
  id: number;
  application_id: number;
  requirement_id?: number | null;
  original_status?: string | null;
  new_status: string;
  reason: string;
  hr_user_id: number;
  created_at: string;
}

export interface AuditDefenseQueryRequest {
  question: string;
}

export interface AuditDefenseQueryResponse {
  application_id: number;
  question: string;
  answer: string;
  referenced_requirements: string[];
  referenced_evidence_ids: number[];
  referenced_overrides: any[];
  assessment_finding?: string | null;
}

export const auditApi = {
  async getAuditTrail(applicationId: number): Promise<FinalAnalysisResponse> {
    return apiClient<FinalAnalysisResponse>(`/audit/${applicationId}`);
  },

  async recordOverride(applicationId: number, payload: AuditOverridePayload): Promise<AuditOverrideResponse> {
    return apiClient<AuditOverrideResponse>(`/audit/${applicationId}/override`, {
      method: 'POST',
      body: JSON.stringify(payload),
    });
  },

  async askAuditDefense(applicationId: number, question: string): Promise<AuditDefenseQueryResponse> {
    return apiClient<AuditDefenseQueryResponse>(`/audit/${applicationId}/ask`, {
      method: 'POST',
      body: JSON.stringify({ question }),
    });
  },
};
